packages/amplify-graphql-auth-transformer/src/resolvers/helpers.ts

```diff
@@ -252,6 +248,58 @@ export const generateOwnerClaimExpression = (ownerClaim: string, refName: string
   return compoundExpression(expressions);
 };
 
+/**
+ * Concatenates multiple owner claims if any
+ */
+export const generateOwnerMultiClaimExpression = (ownerClaim: string, refName: string): Expression => {
+  let expression: Expression;
+  const identityClaims = ownerClaim.split(IDENTITY_CLAIM_DELIMITER);
+  const hasMultiIdentityClaims = identityClaims.length > 1;
+
+  if (hasMultiIdentityClaims) {
+    const additionalClaims: string[] = [];
+    identityClaims.forEach((claim, idx) => {
+      if (idx > 0) {
+        additionalClaims.push(`$currentClaim${idx}`);
```

Comment:
I believe initially we only had `ownerClaim` which was then extended to accommodate multiple claims and these are named as `currentClaim`s. The `ownerClaim` is different from `currentClaim`s because when constructing the multi-claim, we update `ownerClaim` to `ownerClaim :: currentClaim0 :: currentClaim1` etc.